iota/commands/core/get_missing_transactions.py

```diff
@@ -0,0 +1,47 @@
+# coding=utf-8
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+import filters as f
+
+from iota import TransactionHash
+from iota.commands import FilterCommand, RequestFilter, ResponseFilter
+from iota.filters import Trytes
+
+__all__ = [
+    'GetMissingTransactionsCommand',
+]
+
+
+class GetMissingTransactionsCommand(FilterCommand):
+    """
+    Executes `getMissingTransactions` command.
+
+    See :py:meth:`iota.api.StrictIota.get_missing_transactions`.
+    """
+    command = 'getMissingTransactions'
+
+    def get_request_filter(self):
+        return GetMissingTransactionsRequestFilter()
+
+    def get_response_filter(self):
+        return GetMissingTransactionsResponseFilter()
+
+
+class GetMissingTransactionsRequestFilter(RequestFilter):
+    def __init__(self):
+        # ``getMissingTransactions`` does not accept any parameters.
+        # Using a filter here just to enforce that the request is empty.
+        super(GetMissingTransactionsRequestFilter, self).__init__({})
+
+
+class GetMissingTransactionsResponseFilter(ResponseFilter):
+    def __init__(self):
+        super(GetMissingTransactionsResponseFilter, self).__init__({
+            'hashes':
```

Comment:
hey, I have a question about response filters. Why is the duration field never filtered in responses?

Reply:
Good question.  The approach I went for with these was to be restrictive towards requests (because those are within the developer's control) and permissive towards responses (because those are up to the IRI node and out of the developer's control).

Response filters generally only do basic conversions (e.g., for addresses, transaction hashes, etc.); I try to avoid situations where a response from the node would generate an exception in the response filter (principle of least astonishment).

Reply:
That said, I can see the other side of the argument — if the response from the node contains missing/unexpected values, that would be more astonishing than if the lib raised an exception \[that could be caught and handled accordingly in the app\].

For the same reason, it's never really sat well with me that `get_bundles()` doesn't make the validation optional (done this way to match the behaviour of the JS lib at the time, but my preference would be to disable validation by default unless the caller explicitly requests it).

I've never made any of this explicit in the code, so might be good to make a decision one way or the other, and then document it somewhere.